### PR TITLE
[ENH] `GroupbyCategoryForecaster` for applying panel forecasting by category or segment

### LINKED ISSUE #7050: [ENH] TransformSelectForecaster could be able to fit global/panel forecasters
**Describe the solution you'd like**
There is a paper [1] that indicates that the performance of global models (such as a regressor with reduction) is enhanced when a different model is fitted by clusters of similar timeseries.  Currently `TransformSelectForecaster` cannot do that.

**Describe alternatives you've considered**
`TransformSelectForecaster` has mtype `pd.DataFrame` and the categorizer is fit-transformed internally for each series. We could change the mtype to multiindex, apply the categorizer to the panel and then fit-predict the forecasters.

**Additional context**

[1] Semenoglou, Artemios-Anargyros, et al. "Investigating the accuracy of cross-learning time series forecasting methods." International Journal of Forecasting 37.3 (2021): 1072-1084.

### PR DESCRIPTION
**Pull request by felipeangelimvieira**
#### Reference Issues/PRs
Closes  #7050

#### What does this implement/fix? Explain your changes.
Implements a composite that allows users to fit-predict different forecasters for each group, where the groups are assigned according to a series-to-primitive transformation.

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### What should a reviewer concentrate their feedback on?

1. Is this name ok?
2. Is the implemented test enough?

#### Did you add any tests for the change?

Yes, would be happy to hear feedback on the test too.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
- [x] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [x] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [x] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [x] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

File: sktime/forecasting/compose/_grouped.py [hunk 1]
```diff
@@ -122,3 +131,554 @@ def get_test_params(cls, parameter_set="default"):
...
+    >>> group_forecaster = group_forecaster.fit(generated_data, fh=50)
+    >>> #print(f"The chosen category is: {group_forecaster.category}")
+
+    >>> # Print out the predicted value over the given forecasting horizon!
+    >>> # print(group_forecaster.predict(fh=50, X=None))
+    """
+
+    _tags = {
+        "y_inner_mtype": [
+            "pd.DataFrame",
```
Comment (fkiraly) [inline sktime/forecasting/compose/_grouped.py:212]: I think the faiure is due to not handling the case of a single time series correctly.

If you want to deal only with MultiIndex, you can remove the non-multiindexed data types, and afaik the boilerplate will add a dummy level with a single value.

Comment (felipeangelimvieira) [reply sktime/forecasting/compose/_grouped.py:212]: oh perfect. Will do that

Comment (fkiraly) [reply sktime/forecasting/compose/_grouped.py]: oh, ouch, it turns out I misremembered - there is some code duplication between `BaseTransformer` and `BaseForecaster`, but the feature where a single series is coerced to a panel is only present in the duplicate location in `BaseTransformer`, which is ahead of the `BaseForecaster` that does not have this.

I would hence suggest:

* open an issue on the lack of boilerplate functionality in this csae
* add back `"pd.DataFrame"` to the internal types and deal with it somehow - worst case one could copy the code from the `TransformSelectForecaster`

Comment (felipeangelimvieira) [reply sktime/forecasting/compose/_grouped.py]: hmm I understand. I will handle this case in the GroupbyCategoryForecaster class then.

**Pull request merged**